### PR TITLE
MacOSX, FreeBSD support

### PR DESCRIPTION
**Pull request by lu-zero**
Here some initial set of changes to try to have the vc2hqdecode build on other unix-like systems.

### CODE REVIEW

**Comment by lu-zero** [comment]
I locally rebased set, all this contribution should be considered `MIT` licensed and can be relicensed as BBC sees fit.

**Comment by lu-zero** [comment]
Now the tests build on OSX as as well.

**Comment by jamesba** [comment]
This has a problem when building out of tree. The lines you've added in testsuite/Makefile.am use $(top_srcdir) when they should use $(top_builddir).

File: testsuite/Makefile.am [hunk 1]
```diff
@@ -10,7 +10,10 @@ AM_CXXFLAGS = $(VC2HQDECODE_CFLAGS) \
 
 AM_LDFLAGS = $(VC2HQDECODE_LDFLAGS)
 
-LDADD = $(VC2HQDECODE_LIBS)
+LDADD = $(top_srcdir)/vc2inversetransform_sse4_2/libvc2invtransform-sse4-2.la \
```
Comment (jamesba) [inline testsuite/Makefile.am]: This should be top_builddir

Comment (lu-zero) [reply testsuite/Makefile.am]: The comment got lost, updating now.

**Pull request merged**